src/service/mercury/helpers/transformers.ts

```diff
@@ -145,4 +147,941 @@ const transformAccountBalances = async (
+    edges: {
+      node: {
+        amount: string;
+        minAmountA: string;
+        minAmountB: string;
+        source: string;
+        sourceMuxed: string;
+      };
+    }[];
+  };
+}
+
+const transformAccountHistory = async (
+  rawResponse: OperationResult<MercuryAccountHistory>,
+  pubKey: string,
+  network: NetworkNames,
+  getTokenDetails: any // todo
+): Promise<Partial<Horizon.ServerApi.OperationRecord>[]> => {
+  const transferFromEdges = rawResponse.data?.transferFromEvent.edges || [];
+  const transferFrom = await Promise.all(
+    transferFromEdges.map(async (edge) => {
+      const tokenDetails = await getTokenDetails(
+        pubKey,
+        edge.node.contractId,
+        network
+      );
+      const amountBigInt = scValToNative(
+        xdr.ScVal.fromXDR(edge.node.data, "base64")
+      ) as BigInt;
+      return {
+        source_account: pubKey,
+        asset_issuer: edge.node.contractId,
+        asset_code: tokenDetails.symbol,
+        type: "invoke_host_function",
+        type_i: 24,
+        contractId: edge.node.contractId,
+        fnName: scValToNative(xdr.ScVal.fromXDR(edge.node.topic1, "base64")),
+        to: scValToNative(xdr.ScVal.fromXDR(edge.node.topic3, "base64")),
+        from: scValToNative(xdr.ScVal.fromXDR(edge.node.topic2, "base64")),
+        amount: amountBigInt.toString(),
+      } as Partial<Horizon.ServerApi.InvokeHostFunctionOperationRecord>;
+    })
+  );
+
+  const transferToEdges = rawResponse.data?.transferToEvent.edges || [];
+  const transferTo = await Promise.all(
+    transferToEdges.map(async (edge) => {
+      const tokenDetails = await getTokenDetails(
+        pubKey,
+        edge.node.contractId,
+        network
+      );
+      const amountBigInt = scValToNative(
+        xdr.ScVal.fromXDR(edge.node.data, "base64")
+      ) as BigInt;
+      return {
+        source_account: pubKey,
+        asset_issuer: edge.node.contractId,
+        asset_code: tokenDetails.symbol,
+        type: "invoke_host_function",
+        type_i: 24,
+        contractId: edge.node.contractId,
+        fnName: scValToNative(xdr.ScVal.fromXDR(edge.node.topic1, "base64")),
+        to: scValToNative(xdr.ScVal.fromXDR(edge.node.topic3, "base64")),
+        from: scValToNative(xdr.ScVal.fromXDR(edge.node.topic2, "base64")),
+        amount: amountBigInt.toString(),
+      } as Partial<Horizon.ServerApi.InvokeHostFunctionOperationRecord>;
+    })
+  );
+
+  const mintEdges = rawResponse.data?.mintEvent.edges || [];
+  const mint = await Promise.all(
+    mintEdges.map(async (edge) => {
+      const tokenDetails = await getTokenDetails(
+        pubKey,
+        edge.node.contractId,
+        network
+      );
+      const amountBigInt = scValToNative(
+        xdr.ScVal.fromXDR(edge.node.data, "base64")
+      ) as BigInt;
+      return {
+        source_account: pubKey,
+        asset_issuer: edge.node.contractId,
+        asset_code: tokenDetails.symbol,
+        type: "invoke_host_function",
```

Comment:
you can use the enum `OperationResponseType` from `stellar-sdk` for these `type`s

Reply:
ah nice, thank you